### DISCUSSION TITLE
If I want to add a global variable at build time, or modify some configuration, what should I do? Which hook to use?

### DISCUSSION THREAD

**Discussion by lihanspace** [nuxt/nuxt#17393] [Questions] [answered]
If I want to add a global variable at build time, or modify some configuration, what should I do? Which hook to use?

**Comment by Flanker72** [comment, marked as answer]
That's roughly depends on the task you are implementing. The list of available build-time hooks is here: https://github.com/nuxt/framework/blob/main/packages/schema/src/types/hooks.ts#L55

However, maybe implementing a module will help to solve your issue.
